### ISSUE TITLE
Regression: NAT Traversal constantly fails on 1.5.3

### ISSUE THREAD

**Issue by ssg** [ava-labs/avalanchego#915]
**Describe the bug**
NAT traversal constantly fails on 1.5.3 while it's had no problems on Linux pre-1.5.3. NAT traversal with 1.5.2 and older versions work fine from the first attempt. Reverting the container back to 1.5.2 fixes this issue.

This is an upgrade blocker currently.

I feel like this might be a problem with a newer version of goupnp if it's updated. I didn't check the code though.

**To Reproduce**
Start avalanchego v1.5.3.

**Expected behavior**
NAT Traversal should succeed.

**Logs**
1.5.3 logs:
```
ERROR[09-10|02:40:15] /build/nat/nat.go#90: Renewing port mapping try #1 from external port 9651 to internal port 9651 failed with unknown protocol TCP
ERROR[09-10|02:40:16] /build/nat/nat.go#90: Renewing port mapping try #2 from external port 9651 to internal port 9651 failed with unknown protocol TCP
ERROR[09-10|02:40:17] /build/nat/nat.go#90: Renewing port mapping try #3 from external port 9651 to internal port 9651 failed with unknown protocol TCP
WARN [09-10|02:40:18] /build/nat/nat.go#120: Renew NAT Traversal failed from external port 9651 to internal port 9651 with unknown protocol TCP
ERROR[09-10|02:45:18] /build/nat/nat.go#90: Renewing port mapping try #1 from external port 9651 to internal port 9651 failed with unknown protocol TCP
ERROR[09-10|02:45:19] /build/nat/nat.go#90: Renewing port mapping try #2 from external port 9651 to internal port 9651 failed with unknown protocol TCP
ERROR[09-10|02:45:20] /build/nat/nat.go#90: Renewing port mapping try #3 from external port 9651 to internal port 9651 failed with unknown protocol TCP
WARN [09-10|02:45:21] /build/nat/nat.go#120: Renew NAT Traversal failed from external port 9651 to internal port 9651 with unknown protocol TCP
ERROR[09-10|02:50:21] /build/nat/nat.go#90: Renewing port mapping try #1 from external port 9651 to internal port 9651 failed with unknown protocol TCP
ERROR[09-10|02:50:22] /build/nat/nat.go#90: Renewing port mapping try #2 from external port 9651 to internal port 9651 failed with unknown protocol TCP
ERROR[09-10|02:50:23] /build/nat/nat.go#90: Renewing port mapping try #3 from external port 9651 to internal port 9651 failed with unknown protocol TCP
WARN [09-10|02:50:24] /build/nat/nat.go#120: Renew NAT Traversal failed from external port 9651 to internal port 9651 with unknown protocol TCP
ERROR[09-10|02:55:24] /build/nat/nat.go#90: Renewing port mapping try #1 from external port 9651 to internal port 9651 failed with unknown protocol TCP
ERROR[09-10|02:55:25] /build/nat/nat.go#90: Renewing port mapping try #2 from external port 9651 to internal port 9651 failed with unknown protocol TCP
ERROR[09-10|02:55:26] /build/nat/nat.go#90: Renewing port mapping try #3 from external port 9651 to internal port 9651 failed with unknown protocol TCP
WARN [09-10|02:55:27] /build/nat/nat.go#120: Renew NAT Traversal failed from external port 9651 to internal port 9651 with unknown protocol TCP
ERROR[09-10|03:00:27] /build/nat/nat.go#90: Renewing port mapping try #1 from external port 9651 to internal port 9651 failed with unknown protocol TCP
ERROR[09-10|03:00:28] /build/nat/nat.go#90: Renewing port mapping try #2 from external port 9651 to internal port 9651 failed with unknown protocol TCP
ERROR[09-10|03:00:29] /build/nat/nat.go#90: Renewing port mapping try #3 from external port 9651 to internal port 9651 failed with unknown protocol TCP
WARN [09-10|03:00:30] /build/nat/nat.go#120: Renew NAT Traversal failed from external port 9651 to internal port 9651 with unknown protocol TCP
```

1.5.2 log:
```
INFO [09-10|03:08:24] /build/nat/nat.go#74: NAT Traversal successful from external port 9651 to internal port 9651
```

**Operating System**
Linux 3.10.105 x86_64 Intel Braswell CPU

By submitting this issue I agree to the [Terms and Conditions](https://files.avalabs.org/legal/bug_hunt_q2_2020_terms.pdf) of the [Developer Accelerator Program](https://www.avax.network/avalanche-x).

**Comment by ssg** [comment]
The only change regarding NAT I can see is a single line removal of a nil assignment which doesn't seem relevant at all:

https://github.com/ava-labs/avalanchego/compare/v1.5.2...v1.5.3#diff-6adb8c32f9bf431de6b8688047d82e2d9c1db9e8a8e0c35f15b43d4037f6fba0

`goupnp` package doesn't seem to be updated.

However, the whole network code seems to have been overhauled, and golang/net dependency seems to have been upgraded. I don't know if those are related though.

**Comment by ssg** [comment]
This issue has been resolved on my setup by changing the NAT filtering setting from "Secured" to "Open" on my Netgear Orbi.

**Comment by abebeos** [comment]
@ssg @StephenButtolph - can be closed?

**Comment by ssg** [comment]
That PR isn't related to this issue.

**Comment by StephenButtolph** [comment]
I believe this issue should have been resolved in [v1.9.6](https://github.com/ava-labs/avalanchego/releases/tag/v1.9.6) with [e6872d](https://github.com/ava-labs/avalanchego/commit/e6872d0f56fd5bea9588545d1a55c1957a206522). Admittedly, I have absolutely no clue why the setup worked with v1.5.2 though... Regardless I'm going to close this.
